streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -605,7 +607,7 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
         final OptimizableRepartitionNode<K, V> optimizableRepartitionNode = optimizableRepartitionNodeBuilder.build();
         builder.addGraphNode(this.streamsGraphNode, optimizableRepartitionNode);
 
-        return new KStreamImpl<>(builder, repartitionedSourceName, Collections.singleton(repartitionedSourceName), false, optimizableRepartitionNode);
+        return new KStreamImpl<>(repartitionedSourceName, keySerde, valSerde, Collections.singleton(repartitionedSourceName), false, optimizableRepartitionNode, builder);
```

Comment:
Tracking back the passed in `keySerde` and `valSerde` it seems, we would only get user provided serdes via `Joined` but would consider inherited serdes. Or do I miss something?

Reply:
Assuming you meant `but would not consider inherited serdes ..`: hmm, that's a good question. I think it is better to also consider the inherited serdes here before the global configured one. Will fix.

Reply:
Ack, I missed a `not` :)